### DISCUSSION TITLE
FileNotFoundErrors in Python test

### DISCUSSION THREAD

**Discussion by samsrabin** [ESCOMP/CTSM#2059] [Q&A]
Doing `./run_ctsm_py_tests` on Cheyenne gives me 15 errors where it seems to think that the directory I'm running from doesn't exist.

6 of the errors (e.g., for `test_createTestCommand_testfileAndExtraArgs`) are like this:
```
Traceback (most recent call last):
  File "/glade/u/home/samrabin/ctsm_rx_crop_calendars4/python/ctsm/test/test_unit_run_sys_tests.py", line 36, in setUp
    self._original_wd = os.getcwd()
FileNotFoundError: [Errno 2] No such file or directory
```

The other 9 errors (e.g., for `test_check_args_badconfig`) are like this:
```
Traceback (most recent call last):
  File "/glade/u/home/samrabin/ctsm_rx_crop_calendars4/python/ctsm/test/test_unit_subset_data.py", line 34, in setUp
    DEFAULTS_FILE = os.path.join(os.getcwd(), "ctsm/test/testinputs/default_data.cfg")
FileNotFoundError: [Errno 2] No such file or directory
```

Any ideas?

**Comment by root-one-two** [comment]
`os.path.join(os.getcwd(), "ctsm/test/testinputs/default_data.cfg")` here os.getcwd() will take the test path as a root directory and not the source code directory where your script is. 

if you are running the test cases in idea intellij you can go in Edit configuration --> and from working directory remove the test suffix from existing path and try running your test cases

**Reply by samsrabin** [reply]
I'm calling it on the command line from `$CTSMROOT/python`, where the relative path `ctsm/test/testinputs/default_data.cfg` does exist. Also, note that the first error message seems to indicate a problem with `os.getcwd` itself…

**Comment by ekluzek** [comment]
@samsrabin hmmm. I always use the Makefile in the python directory to run run_sys_tests there. That shouldn't matter as long as you are in the python directory, which it sounds like you are. 

One guess is the python version you are using? What does 

python3 --version

give?

Many of the files in the testinputs directory are binary files stored under LFS and you have to do special setup to get them working right. But, this file should be stored directly as an ASCII file directly in git. I guess I'd look at the file to make sure it has content in it...

**Reply by samsrabin** [reply]
`python3 --version` gives `Python 3.7.9`, which seems right. When I start a python3 session and do `os.getcwd()`, it does produce `/glade/u/home/samrabin/ctsm_rx_crop_calendars4/python`, so I really don't know what `test_unit_run_sys_tests.py` is complaining about…

**Reply by samsrabin** [reply]
Oh, and default_data.cfg does contain data and looks uncorrupted.

**Reply by ekluzek** [reply]
That all sounds right to me. I'm not sure what else to suggest, other than let's get meet and look at it together. Something different in your environment causing issues? I'm not sure what though...
